pilot/pkg/model/push_context_test.go

```diff
@@ -237,6 +237,7 @@ func TestEnvoyFilters(t *testing.T) {
 func TestServiceIndex(t *testing.T) {
 	g := NewWithT(t)
 	env := &Environment{}
+	env.SetLedger(&DisabledLedger{})
```

Comment:
is it possible to make the zero value significant so we don't need to do this in every test? Not sure if it is or isn't since its an interface

Reply:
Zero value has to be nil, but we have a few options, aside from what is already there:

1. Move env initialization to a function, instead of struct initializer, and set the value there
2. Make functions leveraging the ledger nil tolerant.

Thoughts?

Reply:
> Make functions leveraging the ledger nil tolerant.

are there a lot of call sites? if its a couple probably makes sense, otherwise maybe a function